drivers/lpsxxx/lpsxxx.c

```diff
@@ -149,11 +152,13 @@ int lpsxxx_read_temp(const lpsxxx_t *dev, int16_t *temp)
 
     DEBUG("[lpsxxx] read_temp: raw data %08" PRIx32 "\n", (uint32_t)val);
 
-    /* compute actual temperature value in °C */
-    res += ((float)val) / TEMP_DIVIDER;
+    /* convert val to c°C */
+    val *= 100;
```

Comment:
Could it be that since val is int16 there's an overflow when multiplying by 100 ?

Reply:
> Could it be that since val is int16 there's an overflow when multiplying by 100 ?

Yes that's it. With the following diff, I get correct temperature values:

```diff
diff --git a/drivers/lpsxxx/lpsxxx.c b/drivers/lpsxxx/lpsxxx.c
index e0047c6cae..7019a94b42 100644
--- a/drivers/lpsxxx/lpsxxx.c
+++ b/drivers/lpsxxx/lpsxxx.c
@@ -131,7 +131,7 @@ int lpsxxx_init(lpsxxx_t *dev, const lpsxxx_params_t * params)
 int lpsxxx_read_temp(const lpsxxx_t *dev, int16_t *temp)
 {
     uint8_t tmp;
-    int16_t val = 0;
+    int32_t val = 0;
     uint16_t res = TEMP_BASE;      /* reference value -> see datasheet */
 
     i2c_acquire(DEV_I2C);
```

Reply:
PRs are always welcome :)